### PR TITLE
Fix #2712: Fix null safety issue in ScalaRunTime.isArray implementation

### LINKED ISSUE #2712: ScalaRunTime.isArray is buggy for instances of JavaScript classes
cf the following [discussion](https://gitter.im/scala-js/scala-js?at=58715fafc02c1a3959fb8c01).

### PR DESCRIPTION
**Pull request by julienrf**
Fixes #2712.

### CODE REVIEW

File: test-suite/js/src/test/scala/org/scalajs/testsuite/scalalib/ScalaRunTimeJSTest.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  @Test def ScalaRunTime_isArray_should_not_fail_with_JS_objects(): Unit = {
+    def isScalaArray(x: Any): Boolean =
+      x match {
+        case _: Array[t] => true
+        case _ => false
+      }
+
+    assertFalse(isScalaArray(js.Array(1, 2, 3)))
+    assertFalse(isScalaArray(js.Dynamic.newInstance(js.Dynamic.global.RegExp)("abc")))
+    assertFalse(isScalaArray(js.Dynamic.literal()))
```
Comment (julienrf) [inline test-suite/js/src/test/scala/org/scalajs/testsuite/scalalib/ScalaRunTimeJSTest.scala]: The above tests were crashing before my modifications in `ScalaRunTime`.

File: test-suite/js/src/test/scala/org/scalajs/testsuite/scalalib/ScalaRunTimeJSTest.scala [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import scala.scalajs.js
+
+class ScalaRunTimeJSTest {
+
+  @Test def ScalaRunTime_isArray_should_not_fail_with_JS_objects(): Unit = {
+    def isScalaArray(x: Any): Boolean =
+      x match {
+        case _: Array[t] => true
+        case _ => false
+      }
```
Comment (sjrd) [inline test-suite/js/src/test/scala/org/scalajs/testsuite/scalalib/ScalaRunTimeJSTest.scala]: Coding style:
* Either start `x match {` on the right of the `=`, or you need an additional pair of `{}` around the entire body.
* Vertically align the `=>` of a `match` if `case`s are one-liners.
* `t` is unused and so it should be `_`

More information: https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md

File: test-suite/js/src/test/scala/org/scalajs/testsuite/scalalib/ScalaRunTimeJSTest.scala [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    assertFalse(isScalaArray(js.Array(1, 2, 3)))
+    assertFalse(isScalaArray(js.Dynamic.newInstance(js.Dynamic.global.RegExp)("abc")))
```
Comment (sjrd) [inline test-suite/js/src/test/scala/org/scalajs/testsuite/scalalib/ScalaRunTimeJSTest.scala]: Any reason you're not using `new js.RegExp("abc")` here?

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/scalalib/ScalaRunTimeTest.scala [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import scala.concurrent.Future
+
+class ScalaRunTimeTest {
+
+  @Test def ScalaRunTime_isArray(): Unit = {
+    def isScalaArray(x: Any): Boolean =
+      x match {
+        case _: Array[t] => true
+        case _ => false
+      }
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/scalalib/ScalaRunTimeTest.scala]: Same remarks on the coding style, obviously.

**Comment by sjrd** [comment]
Also, regarding commit messages:
* Could you remove `#2712` from the first commit, which is not about #2712 per se.
* Start the commit message of the second commit with `Fix #2712: Fix null safety ...` which is [our format for commits addressing an issue](https://github.com/scala-js/scala-js/blob/master/CONTRIBUTING.md#first-line-of-the-commit-message)

**Comment by julienrf** [comment]
I updated the PR according to your remarks.

**Pull request merged**